forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -492,6 +489,16 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
                 // 607.2q linked ability can find cards exiled as cost while it was a spell
                 copied.addExiledCards(c.getExiledCards());
             }
+
+            if (cause != null && cause.isCraft() && toBattlefield) { // retain cards crafted while ETB transformed
+                for (Card craft : cause.getPaidList("Exiled")) {
```

Comment:
it's highly important you don't use LKI list here
> If any of the exiled cards has a characteristic-defining ability that defines its power, that ability will apply. For example, if you somehow used Souls of the Lost to craft Mastercraft Raptor (perhaps because it became a copy of a Dinosaur), its power while in exile is equal to the number of permanent cards in your graveyard, and Mastercraft Raptor's power will change as the number of permanent cards in your graveyard changes. If the characteristic-defining ability can't be applied, then use 0.